--- patches/dev300/apply.orig	2010-11-09 18:38:34.956004724 +0100
+++ patches/dev300/apply	2010-11-09 19:01:20.240004578 +0100
@@ -179,7 +179,7 @@
 [ BuildBits ]
 
 # Upgraded internal ICU to 4.2.1
-icu-4.2.1.diff, cbosdo
+#icu-4.2.1.diff, cbosdo
 
 [ TemporaryHacks ]
 
@@ -878,11 +878,11 @@
 # support server-only versions of 64bit JDKs
 64bit-jdk-server-paths.diff, i#65974, hmth
 # use IBM j9vm over IBM classic JVM
-ibm-j9vm.diff, hmth
+#ibm-j9vm.diff, hmth
 # system db check
 system-db-check.diff, i#65979, hmth
 # User ~/.ooo3 as user dir
-scp2-user-config-ooo3.diff, pmladek
+#scp2-user-config-ooo3.diff, pmladek
 # default inc/lib
 config_office-XINC-XLIB-defaults.diff, i#65976
 # Sandbox problem with wrong path
