--- download_external_sources.sh	2010-07-13 14:11:43.299576221 +0200
+++ download_external_sources.sh	2010-07-13 14:12:21.507575775 +0200
@@ -6,14 +6,16 @@
 export CLONEDIR
 
 if test -e $CLONEDIR/bootstrap/fetch_tarballs.sh -a -e $CLONEDIR/bootstrap/ooo.lst; then
+    sed -e /boost/d -e /seamonkey/d -e /Python/d -e /icu4c/d -e /vigra/d -i $CLONEDIR/bootstrap/ooo.lst
     $CLONEDIR/bootstrap/fetch_tarballs.sh $CLONEDIR/bootstrap/ooo.lst
 else
-    GIT_TAG=`echo $OOO_SOURCEDIRNAME | tr "a-z.-" "A-Z__"`
+    GIT_TAG=$OOO_SOURCEDIRNAME
     tmp=`mktemp -q -d`
 
     cd $tmp
     wget http://cgit.freedesktop.org/libreoffice/bootstrap/plain/fetch_tarballs.sh?id=$GIT_TAG -O fetch_tarballs.sh && chmod 755 fetch_tarballs.sh
     wget http://cgit.freedesktop.org/libreoffice/bootstrap/plain/ooo.lst?id=$GIT_TAG -O ooo.lst
+    sed -e /boost/d -e /seamonkey/d -e /Python/d -e /icu4c/d -e /vigra/d -i ooo.lst
     ./fetch_tarballs.sh ooo.lst &&  rm -f fetch_tarballs.sh ooo.lst
     cd ..
     rmdir $tmp
