--- a/vcl/unx/gtk3/gtk3salnativewidgets-gtk.cxx	2016-09-25 21:35:15.847425158 +0200
+++ b/vcl/unx/gtk3/gtk3salnativewidgets-gtk.cxx	2016-09-25 21:35:41.475426459 +0200
@@ -1382,8 +1382,7 @@
         }
         else
         {
-            //context = mpMenuBarStyle; why does this not work
-            context = gtk_widget_get_style_context(mpWindow);
+            context = mpMenuBarStyle;
         }
         break;
     case CTRL_FIXEDLINE:
