--- a/meson.build	2018-03-10 15:13:37.321248245 +0100
+++ b/meson.build	2018-03-10 15:23:33.461266174 +0100
@@ -57,11 +57,10 @@
         dep_expat = dependency('expat', version: '>=2.2')
         dep_glib = dependency('glib-2.0', version: '>=2.50')
         dep_libsystemd = dependency('libsystemd', version: '>=230')
-        dep_systemd = dependency('systemd', version: '>=230')
 
-        add_project_arguments('-DSYSTEMUIDMAX=' + dep_systemd.get_pkgconfig_variable('systemuidmax'), language: 'c')
-        conf.set('systemunitdir', dep_systemd.get_pkgconfig_variable('systemdsystemunitdir'))
-        conf.set('userunitdir', dep_systemd.get_pkgconfig_variable('systemduserunitdir'))
+        add_project_arguments('-DSYSTEMUIDMAX=999', language: 'c')
+        conf.set('systemunitdir', '/devnull')
+        conf.set('userunitdir', '/devnull')
 endif
 
 #
--- a/src/meson.build	2018-04-14 19:03:59.620501615 +0200
+++ b/src/meson.build	2018-04-14 19:04:10.856501953 +0200
@@ -2,11 +2,6 @@
 # target: subdirs
 #
 
-if use_launcher
-        subdir('units/system')
-        subdir('units/user')
-endif
-
 #
 # target: libbus.so
 #
