--- dbus-c++/configure.ac.orig	2011-10-31 14:48:42.289938724 +0100
+++ dbus-c++/configure.ac	2011-10-31 14:49:12.857940208 +0100
@@ -57,7 +57,7 @@
 	AC_MSG_RESULT(yes)
 
 	AC_DEFINE(GCC_HASCLASSVISIBILITY, 1, [to enable hidden symbols])
-	CXXFLAGS="-fvisibility=hidden"
+	CXXFLAGS="$CXXFLAGS -fvisibility=hidden"
 else
 	AC_MSG_RESULT(no)
 fi 
@@ -129,8 +129,6 @@
 if test "$enable_debug" = "yes" ; then
 	CXXFLAGS="$CXXFLAGS -Wall -ggdb -O0"
 	AC_DEFINE(DEBUG, 1, [Define to enable debug build])
-else
-	CXXFLAGS="$CXXFLAGS -Wall -O3"
 fi
 
 # Doxygen Documentation
