--- patches/dev300/apply
+++ patches/dev300/apply
@@ -1133,7 +1133,7 @@
 # Use the Gnome VFS on KDE as well if it's there (better than nothing)
 vfs-kde-too.diff, michael
 # Trade speed for reliability with valgrind
-valgrind-alloc.diff, michael
+#valgrind-alloc.diff, michael
 
 # disable the IDEA algorithm in the internal openssl; It is pattented.
 # There is no license fee only for non-commercial use.
