--- bin/package-ooo.orig	2010-11-29 20:16:08.700004578 +0100
+++ bin/package-ooo	2010-11-29 20:20:47.972004579 +0100
@@ -212,7 +212,7 @@
 	# shared MIME info
 	mkdir -p $OODESTDIR/usr/share/mime/packages
 	cd $TOOLSDIR/desktop
-	cp openoffice.xml $OODESTDIR/usr/share/mime/packages
+	cp openoffice.xml $OODESTDIR/usr/share/mime/packages/libreoffice.xml
 	if test "z$RUN_POST_INSTALL_SCRIPTS" = "zyes" && 
 	   which update-mime-database >/dev/null 2>&1 ; then
 	    update-mime-database /usr/share/mime || :
