--- patches/dev300/apply.orig	2010-11-05 15:23:33.068004579 +0100
+++ patches/dev300/apply	2010-11-05 15:33:55.500004578 +0100
@@ -14,22 +14,22 @@
 	 Fixes, DefaultSettings, Features, VCL, Misc, HelpContent, \
 	 Icons, CalcFixes, WriterFixes, EasterEgg, \
 	 CWSBackports, Cleanups, WMF, GnomeVFS, WebDAV, \
-	 Layout, VBABits, VBAObjects, CalcErrors, GCJ, \
+	 Layout, VBABits, CalcErrors, GCJ, \
 	 OOXML, OOXMLExportDevel, OOXMLExport, UnitTesting, \
-	 FedoraCommonFixes, LayoutDialogs, Fuzz, \
+	 FedoraCommonFixes, Fuzz, \
 	 BuildFix, WriterDocComparison, \
 	 MySQL, BorderTypes, WriterNavigation
 
 LinuxCommon : Common, Defaults, FontConfigTemporaryHacks, \
 	      FedoraLinuxOnlyFixes, LinuxOnly, msaccess, OpenGLTransitions, \
-	      CliMonoCommon, Mono, MonoComp, AddressBooks, QuickStarter, KDE4, Netbook, \
+	      CliMonoCommon, AddressBooks, QuickStarter, KDE4, Netbook, \
 	      GlobalNSPluginSupport
 MacOSXCommon : Common, Defaults
 Win32Common : Common, Defaults, Win32Only, CliMonoCommon, CliComponent, NotDebian
 # Experimental sections
-Experimental: VBAUntested, ArkOnlyExperimental, \
+Experimental: MonoComp, VBAUntested, ArkOnlyExperimental, \
 	      UnUsedButNotYetRemovedFromGit, WebDAVUpstream, \
-	      PostgreSQL, SELinux, \
+	      PostgreSQL, SELinux, VBAObjects, LayoutDialogs, \
 	      UnitBootstrap, RadioButtons, UnstableLibwpd, WWInProgress, \
 	      KDE4Experimental, MinGW, CalcExperimental, \
 	      OOXMLExportExperimental, CrossWin32Patches, AutoLayout, \
@@ -305,7 +305,7 @@
 
 [ Fixes ]
 # printf arguments cleanup
-buildfix-fmtargs.diff, cabral
+#buildfix-fmtargs.diff, cabral
 
 # use basic (advanced) shapes for rectangle/ellipse on draw/impress's toolbars. make shapes wrap words by default, n#171052, rodo
 sd-toolbar-advanced-shapes.diff, n#171052, rodo
@@ -716,7 +716,7 @@
 
 # disable checkdeliver, no point in doing that, especially when we
 # hardlink to solver instead of copying :-)
-disable-checkdeliver.diff, jholesov
+#disable-checkdeliver.diff, jholesov
 
 [ DefaultSettings ]
 
@@ -793,10 +793,10 @@
 #FIXME src680: separate-helpcontent.diff, jholesov
 
 # use --hash-style=gnu linking when supported
-speed-hash-style.diff, pmladek
+#speed-hash-style.diff, pmladek
 
 # use -Wl,--as-needed, save 1% of relocations
-link-as-needed.diff, i#89511, pmladek
+#link-as-needed.diff, i#89511, pmladek
 
 [ DebianOnly ]
 seamonkey-is-iceape.diff
@@ -874,11 +874,11 @@
 # support server-only versions of 64bit JDKs
 64bit-jdk-server-paths.diff, i#65974, hmth
 # use IBM j9vm over IBM classic JVM
-ibm-j9vm.diff, hmth
+#ibm-j9vm.diff, hmth
 # system db check
 system-db-check.diff, i#65979, hmth
 # User ~/.ooo3 as user dir
-scp2-user-config-ooo3.diff, pmladek
+#scp2-user-config-ooo3.diff, pmladek
 # default inc/lib
 config_office-XINC-XLIB-defaults.diff, i#65976
 # Sandbox problem with wrong path
@@ -1874,7 +1874,7 @@
 [ Fixes ]
 fix-ppt-linespacing-import-export.diff, n#355302, rodo
 
-fields-table-formula.diff, n#631912, cbosdo
+#fields-table-formula.diff, n#631912, cbosdo
 
 # Backport from cbosdo06 CWS
 sw-table-frame-uno-fix.diff, i#112564, cbosdo
@@ -2083,7 +2083,7 @@
 mysqlcppconn-mysqlclient-SONAME.diff
 
 [ BuildBits ]
-novell-win32-agfa-monotype-fonts.diff, tml
+#novell-win32-agfa-monotype-fonts.diff, tml
 
 [ Fixes ]
 # FIXME dev300-m77 sd-print-fix-pageoffset.diff, n#537931, rodo # the code has moved to sd/source/ui/view/DocumentRenderer.cxx
@@ -2129,7 +2129,7 @@
 
 [ BuildBits ]
 # build the extra extensions when enabled
-postprocess-extra-extensions.diff, i#123456, pmladek
+#postprocess-extra-extensions.diff, i#123456, pmladek
 # know how to unpack .oxt files if you want to update a 3rd party one
 solenv-unpack-oxt.diff, pmladek
 
@@ -2300,7 +2300,7 @@
 [ GentooExperimental ]
 SectionOwner => hmth
 # make Python2 optional
-disable-python.diff
+#disable-python.diff
 
 [ NovellEvaluation ]
 # enable the Subscription menu
@@ -2327,8 +2327,8 @@
 [ Features ]
 # embed generic media files inside odf docs, plus various related
 # sound fixes from rodo
-impressmedia-features.diff, i#83753, n#515553, thorsten
-slideshow-sound.diff, n#515553, rodo
+#impressmedia-features.diff, i#83753, n#515553, thorsten
+#slideshow-sound.diff, n#515553, rodo
 
 [ Fixes ]
 vcl-gtk-process-events-on-exit.diff, rodo
--- patches/dev300/jvm-search-path.diff.orig	2010-11-05 15:28:04.940004577 +0100
+++ patches/dev300/jvm-search-path.diff	2010-11-05 15:28:58.880004579 +0100
@@ -1,9 +1,9 @@
 --- configure.in.orig	2009-04-06 15:15:26.000000000 +0200
 +++ configure.in	2009-04-06 15:32:04.000000000 +0200
 @@ -614,6 +614,14 @@
-  
-                           Usage: --with-java-target-version=<jvm version>
- ],,)
+         [Disables the use of Xaw for the Netscape/Mozilla plugin.]),
+ ,)
+ 
 +AC_ARG_WITH(jvm-path,
 +[  --with-jvm-path         Use a specific JVM search path at runtime.
 +
@@ -13,8 +13,8 @@
 +                                 to find JRE/JDK in /usr/lib/jvm/
 +],,)
  AC_ARG_ENABLE(gcjaot,
- [  --enable-gcjaot         Build with[[out]] using Ahead of Time java compilation
-                           support to speed up buildsi by compiling the jars also
+     AS_HELP_STRING([--enable-gcjaot],
+         [Build with[[out]] using Ahead of Time java compilation support to speed
 @@ -3172,6 +3180,26 @@
  fi
  
--- bin/package-ooo.orig	2010-11-05 18:15:03.336004579 +0100
+++ bin/package-ooo	2010-11-05 18:15:32.964004579 +0100
@@ -440,8 +440,8 @@
 }
 
 if ! test -f $OODESTDIR/gid_Module_Root; then
-    echo "Failed to generate package file lists";
-    exit 1;
+    echo "No need to generate package file lists, exit ...";
+    exit 0;
 fi
 
 cd $OODESTDIR
