--- configure.in.orig	2011-04-25 00:04:13.938585334 +0200
+++ configure.in	2011-04-25 00:08:17.430585040 +0200
@@ -4144,6 +4144,11 @@
 if test "$enable_python" != "no"; then
    AC_MSG_RESULT([yes])
    BUILD_TYPE="$BUILD_TYPE PYUNO"
+else
+   AC_MSG_RESULT([no])
+   DISABLE_PYTHON=TRUE
+   AC_SUBST(DISABLE_PYTHON)
+fi
 
 AC_MSG_CHECKING([which python to use])
 if test "$_os" = "WINNT"; then
@@ -4196,12 +4201,6 @@
 AC_SUBST(PYTHON_LIBS)
 HOME=`echo $HOME | $SED 's:\\\\:/:g'`
 AC_SUBST(HOME)
-dnl disable python
-else
-   AC_MSG_RESULT([no])
-   DISABLE_PYTHON=TRUE
-   AC_SUBST(DISABLE_PYTHON)
-fi
 
 dnl ===================================================================
 dnl Check for system translate-toolkit
--- postprocess/packcomponents/makefile.mk	2011-04-17 20:25:50.770857285 +0200
+++ postprocess/packcomponents/makefile.mk	2011-04-17 20:32:17.190857141 +0200
@@ -140,7 +140,6 @@
     placeware \
     preload \
     protocolhandler \
-    pythonloader \
     res \
     rpt \
     rptui \
@@ -193,6 +192,10 @@
     xsltfilter \
     xstor
 
+.IF "$(DISABLE_PYTHON)" != "TRUE"
+my_components += pythonloader
+.ENDIF
+
 .IF "$(OS)" != "WNT" && "$(OS)" != "MACOSX"
 my_components += splash
 .ENDIF
--- scripting/source/pyprov/makefile.mk.orig	2011-04-25 00:05:08.874585333 +0200
+++ scripting/source/pyprov/makefile.mk	2011-04-25 00:24:33.918584452 +0200
@@ -35,7 +35,8 @@
 .INCLUDE :  settings.mk
 
 # --- Targets ------------------------------------------------------
-.IF "$(DISABLE_PYTHON)" != "TRUE"
+# mailmerge is put in an extra services.rdb
+# simply do not fiddle with that. cws sb123, sb129
 ALL : ALLTAR \
     $(DLLDEST)$/officehelper.py	\
         $(DLLDEST)$/mailmerge.py
@@ -65,12 +66,6 @@
 
 .ELSE
 
-.INCLUDE : target.mk
-
-.ENDIF
-
-.ELSE
-
 .INCLUDE : target.mk
 
 .ENDIF
