--- patches/dev300/apply	2009-11-25 18:24:00.000000000 +0100
+++ patches/dev300/apply	2009-11-25 18:33:54.000000000 +0100
@@ -1610,7 +1610,7 @@
 SectionOwner => aprosky
 
 # Allow build to proceed with PaX enabled
-gentoo-pax-fix.diff, aprosky
+#gentoo-pax-fix.diff, aprosky
 # support server-only versions of 64bit JDKs
 64bit-jdk-server-paths.diff, i#65974, hmth
 # use IBM j9vm over IBM classic JVM
