--- solenv/gbuild/extensions/post_AuxTargets.mk	2011-12-04 16:05:33.641023343 +0100
+++ solenv/gbuild/extensions/post_AuxTargets.mk	2011-12-04 16:05:39.937023662 +0100
@@ -76,7 +76,7 @@
 	desktop/scripts/soffice.sh ooo.lst post_download post_download.log \
 	set_soenv set_soenv.last set_soenv.stamp src.downloaded warn
 
-install: build
+install:
 	echo "Installing in $(INSTALLDIR)..." && \
 	ooinstall "$(INSTALLDIR)" && \
 	echo "" && \
