diff --git a/x11-libs/gdk-pixbuf/gdk-pixbuf-2.42.0.ebuild b/x11-libs/gdk-pixbuf/gdk-pixbuf-2.42.0.ebuild
index 4bbdfcaa31a4..616074293776 100644
--- a/x11-libs/gdk-pixbuf/gdk-pixbuf-2.42.0.ebuild
+++ b/x11-libs/gdk-pixbuf/gdk-pixbuf-2.42.0.ebuild
@@ -16,7 +16,6 @@ IUSE="gtk-doc +introspection jpeg tiff"
 # TODO: For windows/darwin support: shared-mime-info conditional, native_windows_loaders option review
 DEPEND="
 	>=dev-libs/glib-2.56.0:2[${MULTILIB_USEDEP}]
-	x11-misc/shared-mime-info
 	>=media-libs/libpng-1.4:0=[${MULTILIB_USEDEP}]
 	jpeg? ( virtual/jpeg:0=[${MULTILIB_USEDEP}] )
 	tiff? ( >=media-libs/tiff-3.9.2:0=[${MULTILIB_USEDEP}] )
