diff --git a/gnome-base/librsvg/librsvg-2.40.21.ebuild b/gnome-base/librsvg/librsvg-2.40.21.ebuild
index 7fa10b848624..9d2267148ec3 100644
--- a/gnome-base/librsvg/librsvg-2.40.21.ebuild
+++ b/gnome-base/librsvg/librsvg-2.40.21.ebuild
@@ -58,7 +58,7 @@ multilib_src_configure() {
 		$(multilib_native_use_enable introspection)
 		$(multilib_native_use_with tools gtk3)
 		$(multilib_native_use_enable vala)
-		--enable-pixbuf-loader
+		$(multilib_native_use_enable !elibc_mingw pixbuf-loader)
 	)
 
 	# -Bsymbolic is not supported by the Darwin toolchain
