diff --git a/dev-libs/libressl/libressl-3.3.1.ebuild b/dev-libs/libressl/libressl-3.3.1.ebuild
index 2592744bafd4..0472ea471773 100644
--- a/dev-libs/libressl/libressl-3.3.1.ebuild
+++ b/dev-libs/libressl/libressl-3.3.1.ebuild
@@ -21,7 +21,7 @@ REQUIRED_USE="test? ( static-libs )"
 
 RDEPEND="!dev-libs/openssl:0"
 DEPEND="${RDEPEND}"
-PDEPEND="app-misc/ca-certificates"
+PDEPEND="!elibc_mingw? ( app-misc/ca-certificates )"
 
 src_prepare() {
 	touch crypto/Makefile.in
