diff --git a/eclass/xdg.eclass b/eclass/xdg.eclass
index 219be712e84d..2c77fcd8d25e 100644
--- a/eclass/xdg.eclass
+++ b/eclass/xdg.eclass
@@ -23,9 +23,10 @@ esac
 
 # Avoid dependency loop as both depend on glib-2
 if [[ ${CATEGORY}/${P} != dev-libs/glib-2.* ]] ; then
+IUSE="+mime"
 DEPEND="
-	dev-util/desktop-file-utils
-	x11-misc/shared-mime-info
+	!elibc_mingw? ( dev-util/desktop-file-utils )
+	mime? ( x11-misc/shared-mime-info )
 "
 fi
 
