--- a/src/systemd/meson.build	2018-03-03 22:56:45.111894767 +0100
+++ b/src/systemd/meson.build	2018-03-03 22:58:09.791897314 +0100
@@ -22,9 +22,6 @@
         sd-daemon.h
         sd-event.h
         sd-id128.h
-        sd-journal.h
-        sd-login.h
-        sd-messages.h
 '''.split()
 
 # https://github.com/mesonbuild/meson/issues/1633
--- a/src/libsystemd/meson.build	2018-03-03 21:35:53.919748869 +0100
+++ b/src/libsystemd/meson.build	2018-03-03 22:47:53.195878770 +0100
@@ -15,8 +15,6 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with systemd; If not, see <http://www.gnu.org/licenses/>.
 
-sd_login_c = files('sd-login/sd-login.c')
-
 libsystemd_sources = files('''
         sd-bus/bus-common-errors.c
         sd-bus/bus-common-errors.h
@@ -58,39 +56,11 @@
         sd-bus/bus-type.h
         sd-bus/sd-bus.c
         sd-daemon/sd-daemon.c
-        sd-device/device-enumerator-private.h
-        sd-device/device-enumerator.c
-        sd-device/device-internal.h
-        sd-device/device-private.c
-        sd-device/device-private.h
-        sd-device/device-util.h
-        sd-device/sd-device.c
         sd-event/sd-event.c
-        sd-hwdb/hwdb-internal.h
-        sd-hwdb/hwdb-util.h
-        sd-hwdb/sd-hwdb.c
         sd-id128/id128-util.c
         sd-id128/id128-util.h
         sd-id128/sd-id128.c
-        sd-netlink/generic-netlink.c
-        sd-netlink/local-addresses.c
-        sd-netlink/local-addresses.h
-        sd-netlink/netlink-internal.h
-        sd-netlink/netlink-message.c
-        sd-netlink/netlink-socket.c
-        sd-netlink/netlink-types.c
-        sd-netlink/netlink-types.h
-        sd-netlink/netlink-util.c
-        sd-netlink/netlink-util.h
-        sd-netlink/rtnl-message.c
-        sd-netlink/sd-netlink.c
-        sd-network/network-util.c
-        sd-network/network-util.h
-        sd-network/sd-network.c
-        sd-path/sd-path.c
-        sd-resolve/sd-resolve.c
-        sd-utf8/sd-utf8.c
-'''.split()) + sd_login_c
+'''.split())
 
 libsystemd_static = static_library(
         'systemd',
