--- src/create_torrent_dialog.cpp	2012-02-18 14:48:27.254215036 +0100
+++ src/create_torrent_dialog.cpp	2012-02-18 14:48:47.246215461 +0100
@@ -273,11 +273,7 @@
 					{
 						throw;
 					}
-				#if M_LT_GET_VERSION() < M_GET_VERSION(0, 16, 0)
-					catch(fs::basic_filesystem_error<fs::path>& e)
-				#else
 					catch(fs::filesystem_error& e)
-				#endif
 					{
 						M_THROW(__(
 							"Error while reading torrent files. File '%1': %2.",
@@ -324,11 +320,7 @@
 					{
 						throw;
 					}
-				#if M_LT_GET_VERSION() < M_GET_VERSION(0, 16, 0)
-					catch(fs::basic_filesystem_error<fs::path>& e)
-				#else
 					catch(fs::filesystem_error& e)
-				#endif
 					{
 						M_THROW(__(
 							"Error while reading torrent files. File '%1': %2.",
