diff --git a/include/libtorrent/ip_filter.hpp b/include/libtorrent/ip_filter.hpp
index 4e47b39a1d..4455624449 100644
--- a/include/libtorrent/ip_filter.hpp
+++ b/include/libtorrent/ip_filter.hpp
@@ -41,10 +41,16 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <vector>
 
 #include <boost/limits.hpp>
-#include <boost/utility.hpp>
 #include <boost/cstdint.hpp>
 #include <boost/tuple/tuple.hpp>
 
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106700
+#include <boost/next_prior.hpp>
+#else
+#include <boost/utility.hpp>
+#endif
+
 #include "libtorrent/aux_/disable_warnings_pop.hpp"
 
 #include "libtorrent/address.hpp"
