diff --git a/src/kademlia/routing_table.cpp b/src/kademlia/routing_table.cpp
index a23500b69f..dfc3500930 100644
--- a/src/kademlia/routing_table.cpp
+++ b/src/kademlia/routing_table.cpp
@@ -54,6 +54,13 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <boost/cstdint.hpp>
 #include <boost/bind.hpp>
 
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106700
+#include <boost/next_prior.hpp>
+#else
+#include <boost/utility.hpp>
+#endif
+
 #include "libtorrent/aux_/disable_warnings_pop.hpp"
 
 using boost::uint8_t;
