--- libevent.c.orig	2016-01-31 23:22:34.947714766 +0100
+++ libevent.c	2016-01-31 23:38:00.019761698 +0100
@@ -99,13 +99,13 @@
 /* }}} */
 
 #define ZVAL_TO_BASE(zval) \
-	(php_event_base_t *)zend_fetch_resource2_ex(zval, "event base", le_event_base, le_event_base)
+	(php_event_base_t *)zend_fetch_resource_ex(zval, "event base", le_event_base)
 
 #define ZVAL_TO_EVENT(zval) \
-	(php_event_t *)zend_fetch_resource2_ex(zval, "event", le_event, le_event)
+	(php_event_t *)zend_fetch_resource_ex(zval, "event", le_event)
 
 #define ZVAL_TO_BEVENT(zval) \
-	(php_bufferevent_t *)zend_fetch_resource2_ex(zval, "buffer event", le_bufferevent, le_bufferevent)
+	(php_bufferevent_t *)zend_fetch_resource_ex(zval, "buffer event", le_bufferevent)
 
 /* {{{ internal funcs */
 
@@ -361,8 +361,11 @@
 
 	base->events = 0;
 
-	base->rsrc_id = zend_list_insert(base, le_event_base);
-	ZVAL_COPY_VALUE(return_value, base->rsrc_id);
+	RETVAL_RES(zend_register_resource(base, le_event_base));
+	base->rsrc_id = return_value;
+	Z_TRY_ADDREF_P(return_value);
+	/*base->rsrc_id = zend_list_insert(base, le_event_base);
+	ZVAL_COPY_VALUE(return_value, base->rsrc_id);*/
 }
 /* }}} */
 
@@ -408,7 +411,7 @@
 		RETURN_FALSE;
 	}
 
-	zend_list_close(Z_RES_P(base->rsrc_id));
+	zend_list_close(Z_RES_P(zbase/*base->rsrc_id*/));
 }
 /* }}} */
 
@@ -579,8 +582,11 @@
 	ZVAL_NULL(&event->stream_id);
 	TSRMLS_SET_CTX(event->thread_ctx);
 
-	event->rsrc_id = zend_list_insert(event, le_event);
-	ZVAL_COPY_VALUE(return_value, event->rsrc_id);
+	RETVAL_RES(zend_register_resource(event, le_event));
+	event->rsrc_id = return_value;
+	Z_TRY_ADDREF_P(return_value);
+	/*event->rsrc_id = zend_list_insert(event, le_event);
+	ZVAL_COPY_VALUE(return_value, event->rsrc_id);*/
 }
 /* }}} */
 
@@ -603,8 +609,8 @@
 		--event->base->events;
 		event->base = NULL;
 	}
-	event_del (event->event);
-	zend_list_close(Z_RES_P(event->rsrc_id));
+	event_del(event->event);
+	zend_list_close(Z_RES_P(zevent/*event->rsrc_id*/));
 }
 /* }}} */
 
@@ -996,8 +1002,11 @@
 
 	TSRMLS_SET_CTX(bevent->thread_ctx);
 
-	bevent->rsrc_id = zend_list_insert(bevent, le_bufferevent);
-	ZVAL_COPY_VALUE(return_value, bevent->rsrc_id);
+	RETVAL_RES(zend_register_resource(bevent, le_bufferevent));
+	bevent->rsrc_id = return_value;
+	Z_TRY_ADDREF_P(return_value);
+	/*bevent->rsrc_id = zend_list_insert(bevent, le_bufferevent);
+	ZVAL_COPY_VALUE(return_value, bevent->rsrc_id);*/
 }
 /* }}} */
 
@@ -1019,7 +1028,7 @@
 		--bevent->base->events;
 		bevent->base = NULL;
 	}
-	zend_list_close(Z_RES_P(bevent->rsrc_id));
+	zend_list_close(Z_RES_P(zbevent/*bevent->rsrc_id*/));
 }
 /* }}} */
 
